4-projects/modules/single_project/main.tf

```diff
@@ -20,15 +20,14 @@ locals {
 }
 
 module "project" {
-  source                      = "terraform-google-modules/project-factory/google"
-  version                     = "~> 10.1"
-  random_project_id           = "true"
-  impersonate_service_account = var.impersonate_service_account
```

Comment:
@bharathkkb ,
 
Yes, can we remove the impersonate_service_account in the `modules/single_project` and also in the `modules/base_env`.
Good catch, Thank you!
 
But we are not able to remove the `terraform_service_account` the in the vars because it is used in the places bellow:
 
`modules/infra_pipelines` 
https://github.com/terraform-google-modules/terraform-example-foundation/blob/9c320d8326c9dcd0799093a985fdb0fc2fe7cf04/4-projects/modules/infra_pipelines/main.tf#L153-L159
 
All providers that are in the environment folder (Ex: `4-projects/business_unit_1/development/providers.tf` ) 
https://github.com/terraform-google-modules/terraform-example-foundation/blob/9c320d8326c9dcd0799093a985fdb0fc2fe7cf04/4-projects/business_unit_1/development/providers.tf#L17-L31